src/stock.rs

```diff
@@ -14,8 +15,14 @@ impl Stock {
     pub async fn load_historical_prices(
         &mut self,
         time_frame: TimeFrame,
+        end_date: DateTime<Utc>,
     ) -> Result<(), Box<dyn Error>> {
-        self.bars = history::retrieve_interval(self.symbol.as_str(), time_frame.interval()).await?;
+        self.bars = if let Some(duration) = time_frame.duration() {
+            history::retrieve_range(self.symbol.as_str(), end_date - duration, Some(end_date))
```

Comment:
We see here that `retrieve_range` is designed with an optional end date in mind. So yeah, that kinda corroborates my earlier point about using a start date instead... :laughing:

Reply:
What we can do is not have a start date at first, but only calculate it the first time the user starts panning to the left.